packages/fx-core/tests/plugins/resource/function/unit/generateArmTemplates.test.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import "mocha";
+
+import * as chai from "chai";
+import chaiAsPromised from "chai-as-promised";
+import fs from "fs-extra";
+import * as path from "path";
+
+import { AzureSolutionSettings, PluginContext } from "@microsoft/teamsfx-api";
+import { FunctionPlugin } from "../../../../../src";
+import { ConstantString, mockSolutionUpdateArmTemplates, ResourcePlugins } from "../../util";
+import {
+  DependentPluginInfo,
+  FunctionPluginInfo,
+} from "../../../../../src/plugins/resource/function/constants";
+
+chai.use(chaiAsPromised);
+
+describe("FrontendGenerateArmTemplates", () => {
+  let functionPlugin: FunctionPlugin;
+  let pluginContext: any;
+
+  beforeEach(() => {
+    functionPlugin = new FunctionPlugin();
+    pluginContext = {
+      configOfOtherPlugins: new Map<string, Map<string, string>>([
+        [
+          DependentPluginInfo.solutionPluginName,
+          new Map<string, string>([
+            [DependentPluginInfo.resourceGroupName, "ut"],
+            [DependentPluginInfo.subscriptionId, "ut"],
+            [DependentPluginInfo.resourceNameSuffix, "ut"],
+          ]),
+        ],
+      ]),
+      app: {
+        name: {
+          short: "ut",
+        },
+      },
+      projectSettings: {
+        appName: "ut",
+        programmingLanguage: "javascript",
+      },
+      config: new Map(),
+      root: path.join(__dirname, "ut"),
+    };
+  });
```

Comment:
This context setup is exactly the same as code in *packages\fx-core\tests\plugins\resource\function\unit\scaffold.test.ts*. Can we move it to utils.ts for test?